cpu/cc2538/include/cc2538_rf.h

```diff
@@ -0,0 +1,321 @@
+/* /TODO */
+
+#define CC2538_MIN_FREQ             (2394)
+#define CC2538_MAX_FREQ             (2507)
+
+#define CC2538_RF_POWER_DEFAULT     (3)    /**< Default output power in dBm */
+#define CC2538_RF_CHANNEL_DEFAULT   (26U)
+#define CC2538_RF_PANID_DEFAULT     (0x0023)
+
+#define OUTPUT_POWER_MIN            (-24)  /**< Min output power in dBm */
+#define OUTPUT_POWER_MAX            (7)    /**< Max output power in dBm */
+#define NUM_POWER_LEVELS            ( OUTPUT_POWER_MAX - OUTPUT_POWER_MIN + 1 )
+
+#define RFCORE_ASSERT(expr) (void)( (expr) || RFCORE_ASSERT_failure(#expr, __FUNCTION__, __LINE__) )
+
+#if DEVELHELP
+#define RFCORE_WAIT_UNTIL(expr) while (!(expr)) {                                        \
+    DEBUG("RFCORE_WAIT_UNTIL(%s) at line %u in %s()\n", #expr, __LINE__, __FUNCTION__);  \
+    thread_yield();                                                                      \
+}
+#else
+#define RFCORE_WAIT_UNTIL(expr) while (!(expr)) thread_yield()
+#endif
+
+#define RFCORE_FLUSH_RECEIVE_FIFO() rfcore_strobe(ISFLUSHRX)
+
+#define ABS_DIFF(x, y)          ( ((x) < (y))? ((y) - (x)) : ((x) - (y)) )
+#define BOOLEAN(x)              ( (x) != 0 )
+#define NOT(x)                  ( (x) == 0 )
+#define GET_BYTE(buffer, index) ( (unsigned char*)(buffer) )[index]
+
+#define BIT(n) ( 1 << (n) )
+
+enum {
+    FSM_STATE_IDLE           =  0,
+    FSM_STATE_RX_CALIBRATION =  2,
+    FSM_STATE_TX_CALIBRATION = 32,
+};
+
+/*
+ * @brief RFCORE_XREG_RFERRM bits
+ */
+enum {
+    STROBE_ERR = BIT(6),
+    TXUNDERF   = BIT(5),
+    TXOVERF    = BIT(4),
+    RXUNDERF   = BIT(3),
+    RXOVERF    = BIT(2),
+    RXABO      = BIT(1),
+    NLOCK      = BIT(0),
+};
+
+ /*
+  * @brief RFCORE_XREG_FRMCTRL0 bits
+  */
+enum {
+    ENERGY_SCAN      = BIT(4),
+    AUTOACK          = BIT(5),
+    AUTOCRC          = BIT(6),
+    APPEND_DATA_MODE = BIT(7),
+};
+
+/*
+ * @brief RFCORE_XREG_RFIRQM0 / RFCORE_XREG_RFIRQF0 bits
+ */
+enum {
+    ACT_UNUSED       = BIT(0),
+    SFD              = BIT(1), /**< Start of frame event */
+    FIFOP            = BIT(2),
+    SRC_MATCH_DONE   = BIT(3),
+    SRC_MATCH_FOUND  = BIT(4),
+    FRAME_ACCEPTED   = BIT(5),
+    RXPKTDONE        = BIT(6), /**< End of frame event */
+    RXMASKZERO       = BIT(7),
+};
+
+/* Values for use with CCTEST_OBSSELx registers: */
+#define OBSSEL_EN BIT(7)
+enum {
+    rfc_obs_sig0 = 0,
+    rfc_obs_sig1 = 1,
+    rfc_obs_sig2 = 2,
+};
+
+/* Values for RFCORE_XREG_RFC_OBS_CTRLx registers: */
+enum {
+    constant_value_0 = 0x00, /**< Constant value */
+    constant_value_1 = 0x01, /**< Constant value */
+    rfc_sniff_data   = 0x08, /**< Data from packet sniffer. Sample data on rising edges of sniff_clk. */
+    rfc_sniff_clk    = 0x09, /**< 250kHz clock for packet sniffer data. */
+    rssi_valid       = 0x0c, /**< Pin is high when the RSSI value has been updated at least once since RX was started. Cleared when leaving RX. */
```

Comment:
I don't want to be too picky but according to our [coding conventions](https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions) we aim for no more than 80 characters per line, the absolute maximum should be 100 characters per line.